### PR TITLE
Tests endpoints firmware and chargepoint

### PR DESCRIPTION
**Pull request by Idhao**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1206575137133723/1206645935695266) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by tvelu77 — APPROVED** [review]
The test seems good to me, nice job !

File: src/main/java/fr/uge/chargepointconfiguration/controller/ChargepointController.java [hunk 1]
```diff
@@ -49,8 +49,7 @@ public ChargepointController(ChargepointRepository chargepointRepository) {
           })
   @GetMapping(value = "/chargepoint/all")
   public List<Chargepoint> getAllChargepoints() {
-    return StreamSupport.stream(chargepointRepository.findAll().spliterator(), false)
-            .collect(Collectors.toList());
+    return chargepointRepository.findAll();
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/controller/ChargepointController.java:52]: prefer the use of DTOs

File: src/main/java/fr/uge/chargepointconfiguration/controller/FirmwareController.java [hunk 2]
```diff
@@ -47,9 +47,8 @@ public FirmwareController(FirmwareRepository firmwareRepository) {
           mediaType = "application/json",
           schema = @Schema(implementation = Firmware.class)) })
   @GetMapping(value = "/firmware/all")
-  public List<Firmware> getAllChargepoints() {
-    return StreamSupport.stream(firmwareRepository.findAll().spliterator(), false)
-        .collect(Collectors.toList());
+  public List<Firmware> getAllFirmwares() {
+    return firmwareRepository.findAll();
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/controller/FirmwareController.java:51]: prefere the use of DTOs

File: src/main/java/fr/uge/chargepointconfiguration/controller/FirmwareController.java [hunk 3]
```diff
@@ -71,7 +70,7 @@ public List<Firmware> getAllChargepoints() {
           content = @Content)
   })
   @GetMapping(value = "/firmware/{id}")
-  public Optional<Firmware> getChargepointById(
+  public Optional<Firmware> getFirmwareById(
       @Parameter(description = "id of firmware to be searched") @PathVariable int id) {
     return firmwareRepository.findById(id);
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/controller/FirmwareController.java:75]: prefere the use of DTOs

File: src/main/java/fr/uge/chargepointconfiguration/repository/ChargepointRepository.java [hunk 4]
```diff
@@ -21,4 +23,12 @@ public interface ChargepointRepository extends CrudRepository<Chargepoint, Integ
...
   Chargepoint findBySerialNumberChargepointAndConstructor(String serialNumber,
                                                           String constructor);
+
+
+  /**
+   * Return a list of registered Chargepoints from database.
+   *
+   * @return A list of Chargepoints or an empty list if no chargepoints are registered.
+   */
+  List<Chargepoint> findAll();
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/repository/ChargepointRepository.java:33]: good !

File: src/test/java/fr/uge/chargepointconfiguration/controller/ChargepointControllerTest.java
Comment (Adrien-COSTANDI) [inline src/test/java/fr/uge/chargepointconfiguration/controller/ChargepointControllerTest.java:1]: ask about the relevance of the tests

**Review by Adrien-COSTANDI — CHANGES_REQUESTED** [review]
need some DTOs and check about the tests

**Pull request closed without merging**